check_api/src/main/java23/com/google/errorprone/ConstantStringExpressions.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone;
+
+import com.google.errorprone.VisitorState;
+import com.sun.tools.javac.util.Convert;
+
+/**
+ * @see VisitorState#getConstantExpression(Object)
+ */
+final class ConstantStringExpressions {
+  private ConstantStringExpressions() {}
+
+  static String toConstantStringExpression(Object value, VisitorState state) {
+    return state
+        .getElements()
+        .getConstantExpression(
+            value instanceof CharSequence charSequence ? charSequence.toString() : value);
```

Comment:
Here I did keep the `instanceof` pattern matching, because this code anyway only applies to JDK 23+, but if that anyway impacts the Google-internal setup, that can be changed, of course.